### PR TITLE
Advanced group search fix master

### PR DESCRIPTION
**Pull request by markdoub**
This fixes the problems in the related resources page caused by advanced search

### CODE REVIEW

**Comment by azerbini** [comment]
@markdoub & @ahaith  please use this PR to add further improvements to the advanced search, esp. #20 and #21

File: ENV/lib/python2.7/site-packages/arches/app/views/search.py [hunk 1]
```diff
@@ -192,6 +192,14 @@ def build_search_results_dsl(request):
     boolquery = Bool()
     boolfilter = Bool()
     if term_filter != '':
+        is_empty_temporal_filter = True
```
Comment (azerbini) [inline ENV/lib/python2.7/site-packages/arches/app/views/search.py]: @markdoub  this bit still gives problems with the Related Resources search: 
screen shot 2017-10-06 at 16 31 32

Comment (markdoub) [reply ENV/lib/python2.7/site-packages/arches/app/views/search.py]: Sorry, this doesn't look right. Line 194 should not be there...

Comment (azerbini) [reply ENV/lib/python2.7/site-packages/arches/app/views/search.py]: @markdoub  
```
        if len(temporal_filter['year_min_max']) != 0:
            is_empty_temporal_filter = False
        if len(temporal_filter['filters']) != 0:
            is_empty_temporal_filter = False

```

works on Rel Resources search, but haven't tested it platform-wide

File: ENV/lib/python2.7/site-packages/arches/app/views/search.py [hunk 2]
```diff
@@ -191,13 +191,12 @@ def build_search_results_dsl(request):
...
-    if term_filter != '':
-        is_empty_temporal_filter = True
-        for index, time_filter in enumerate(temporal_filter):
-            if len(time_filter['year_min_max']) != 0:
-                is_empty_temporal_filter = False
-            if len(time_filter['filters']) != 0:
-                is_empty_temporal_filter = False
+    is_empty_temporal_filter = True
+    for index, time_filter in enumerate(temporal_filter):
+        if len(time_filter['year_min_max']) != 0:
```
Comment (azerbini) [inline ENV/lib/python2.7/site-packages/arches/app/views/search.py]: Problem persists on ``` if len(time_filter['year_min_max']) != 0:``` (string indices must be integers)

Comment (azerbini) [reply ENV/lib/python2.7/site-packages/arches/app/views/search.py]: @markdoub

**Comment by azerbini** [comment]
@markdoub, [b913eec](https://github.com/azerbini/eamena_v3/pull/15/commits/b913eec0273795aaf2d8a0ee48d4aab236abe875) does fix the previous issue on Related Resource search, but searching (whether by Attirbute or Time) in Related Resources is still not working. 

I'd suggest that you actually get rid of the different scenario and implement in Related Resources the same logic that you have for the Advanced Search. The two searches should be identical.

**Comment by azerbini** [comment]
https://github.com/azerbini/eamena_v3/pull/15/commits/44191b922fe00837f68ebd6a563a1152673a0290 , https://github.com/azerbini/eamena_v3/pull/15/commits/354be60129fa182fc4ccc95fa63291b2e45d6115 , https://github.com/azerbini/eamena_v3/pull/15/commits/8200c47b93c8b855b5634af6e0361cbb2db3170a , https://github.com/azerbini/eamena_v3/pull/15/commits/ed138eb878e6f93005b31cce9e1d4d0d8c1f9043 , https://github.com/azerbini/eamena_v3/pull/15/commits/72abd6214ac2a990af24da71c5dd2f635839e2a8 all address issues #20 and #44

**Pull request merged**